src/sentry/workflow_engine/endpoints/validators/base/data_condition.py

```diff
@@ -1,30 +1,77 @@
+from abc import abstractmethod
+from typing import Any, Generic, TypeVar
+
+from jsonschema import ValidationError as JsonValidationError
 from rest_framework import serializers
-from rest_framework.fields import Field
 
 from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.utils.registry import NoRegistrationExistsError
+from sentry.workflow_engine.endpoints.validators.utils import validate_json_schema
+from sentry.workflow_engine.models.data_condition import CONDITION_OPS, Condition
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler
+
+ComparisonType = TypeVar("ComparisonType")
+ConditionResult = TypeVar("ConditionResult")
+
+
+class AbstractDataConditionValidator(
+    CamelSnakeSerializer,
+    Generic[ComparisonType, ConditionResult],
+):
+    id = serializers.IntegerField(required=False)
+    type = serializers.ChoiceField(choices=[(t.value, t.value) for t in Condition])
+    comparison = serializers.JSONField(required=True)
+    condition_result = serializers.JSONField(required=True)
+    condition_group_id = serializers.IntegerField(required=True)
+
+    @abstractmethod
+    def validate_comparison(self, value: Any) -> ComparisonType:
+        pass
+
+    @abstractmethod
+    def validate_condition_result(self, value: Any) -> ConditionResult:
+        pass
+
 
+class BaseDataConditionValidator(
+    AbstractDataConditionValidator[Any, Any],
+):
```

Comment:
just to make sure i understand, this validator is for any condition that is not one of the `CONDITION_OPS`, conditions that use something in `CONDITION_OPS` will implement `AbstractDataConditionValidator` like `MetricAlertComparisonConditionValidator`?

Reply:
💯 yeah - yuuup. this `BaseDataConditionValidator` will respond to any custom handler. if we don't have a custom handler and it's just a condition op, you only need to setup the validator in the API layer using this.

PS, i am planning to change the `BaseDataConditionValidator` name to `DataConditionValidator` in a future pr to help with any confusion around it. `Base` reads like it should be extended to me, and in this case, it's the generic implementation. i figured the diff was big enough on this PR already though.